src/FakeClient/GraphicsCaptureEventProcessor.h

```diff
@@ -0,0 +1,268 @@
+
+      // If the interval don't overlap just add the length of the interval to the frame time, else
+      // if there's overlap just add the new part of the interval that haven't been taking into
+      // account.
+      if (begin_timestamp_ns >= current_range_end) {
+        frame_time_ns += (end_timestamp_ns - begin_timestamp_ns);
+        current_range_end = end_timestamp_ns;
+      } else if (end_timestamp_ns > current_range_end) {
+        frame_time_ns += end_timestamp_ns - current_range_end;
+        current_range_end = end_timestamp_ns;
+      }
+    }
+
+    return frame_time_ns;
+  }
+
+  void CalculateCpuStats() {
+    LOG("Calculating CPU Times");
+
+    uint64_t frame_boundary_count = frame_start_boundary_timestamps_.size();
+    FAIL_IF(frame_boundary_count <= 2,
+            "Not enough calls to vkQueuePresentKHR to calculate CPU frame times.");
+
+    CalculateCpuFrameDurations();
+    GenerateCpuDurationDistribution();
+    CalculateCpuAvgFrameTime();
+  }
+
+  void CalculateCpuFrameDurations() {
+    for (size_t current_frame = 1; current_frame < frame_start_boundary_timestamps_.size();
+         ++current_frame) {
+      uint64_t frame_time_ns = (frame_start_boundary_timestamps_[current_frame] -
+                                frame_start_boundary_timestamps_[current_frame - 1]);
+      frame_cpu_durations_ns_.emplace_back(frame_time_ns);
+    }
+  }
+
+  void GenerateCpuDurationDistribution() {
+    cpu_time_distribution_.fill(0u);
+    for (uint64_t frame_duration_ns : frame_cpu_durations_ns_) {
+      UpdateFrameDurationDistribution(frame_duration_ns, cpu_time_distribution_);
+    }
+  }
+
+  void CalculateCpuAvgFrameTime() {
+    uint64_t total_duration_ns = std::accumulate(frame_cpu_durations_ns_.begin(),
+                                                 frame_cpu_durations_ns_.end(), uint64_t(0));
+    cpu_avg_frame_time_ms_ = (static_cast<double>(total_duration_ns) / 1.0e6) /
+                             static_cast<double>(frame_start_boundary_timestamps_.size() - 1);
+  }
+
+  template <typename Distribution>
+  static void UpdateFrameDurationDistribution(uint64_t frame_time_ns, Distribution& distribution) {
+    if (frame_time_ns > 0) {
+      double frame_time_ms = frame_time_ns / 1.0e6;
+      int distribution_index = std::min(static_cast<int>(frame_time_ms), kMaxTimeMs);
+      ++distribution[distribution_index];
+    }
+  }
+
+  template <typename Distribution>
+  static void WriteToCsvFile(absl::string_view filename, const Distribution& distribution,
+                             const double avg, const size_t num_frames) {
+    constexpr int kCentiles = 100;
+    std::string header = "num_frames,avg_ms_per_frame";
+    for (int centile = 1; centile <= kCentiles; ++centile) {
+      absl::StrAppend(&header, absl::StrFormat(",%d_%dtile_ms_per_frame", centile, kCentiles));
+    }
+    std::string output;
+    absl::StrAppend(&output, header, "\n", absl::StrFormat("%u,%.2f", num_frames, avg));
```

Comment:
I suggest splitting this into two lines, one for the "\n" and one for the "%u,%.2f", to clarify where the generation of the header ends.

Reply:
done